src/framework/audio/internal/rpc/rpccontroller.cpp

```diff
@@ -0,0 +1,160 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2021 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#include "rpccontroller.h"
+
+#include "log.h"
+
+using namespace mu::audio;
+using namespace mu::audio::rpc;
+
+static Target rpcSeqTarget = Target(TargetName::Sequencer);
+
+void RpcController::init()
+{
+    //! NOTE We could use a by target controller factory here,
+    //! but we have not abstract RPC system, but a specialized one for audio, and the targets are known in advance.
+    //! So, a factory is deliberately not used to have everything in one place.
+    rpcChannel()->listen([this](const Msg& msg) {
+        switch (msg.target.name) {
+        case TargetName::Sequencer:
+            sequencerHandle(msg);
+            break;
+        case TargetName::Undefined: {
+            LOGE() << "received message for undefined target, will be ignored";
+        }
+        }
+    });
+}
+
+ISequencerPtr RpcController::sequencer() const
+{
+    return audioEngine()->sequencer();
+}
+
+void RpcController::sequencerHandle(const Msg& msg)
+{
+    static std::map<Method, Call> calls;
+    auto bindMethod = [](std::map<Method, Call>& calls, const Method& method, const Call& call) {
+        calls.insert({ method, call });
+    };
+
+    //! NOTE We could use reflection here so as not to write this code of the same type.
+    //! But it's not that simple. Rpc can work in serialization mode,
+    //! and then some arguments cannot be simply passed to the target class, they should be deserialized first.
+    //! The second problem is subscription to channels for the given argument.
+    //! The target class just has a method to get the channel.
+    //! But to use the channel in RPC, we need to do the same as in bindMidiTickPlayed
+    //! So, it is better to have the preparation of arguments and their conversion in one place.
+
+    if (calls.empty()) {
+        // Bind to sequencer
+
+        bindMethod(calls, "initMIDITrack", [this](const Args& args) {
+            sequencer()->initMIDITrack(args.arg<ISequencer::track_id>(0));
+        });
+
+        bindMethod(calls, "initAudioTrack", [this](const Args& args) {
+            sequencer()->initMIDITrack(args.arg<ISequencer::track_id>(0));
```

Comment:
maybe initAudioTrack instead of initMIDITrack?

Reply:
sure, fixed, thanks!!